open-api/src/testFixtures/java/org/apache/iceberg/rest/RESTCatalogServer.java

```diff
@@ -0,0 +1,125 @@
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.util.component.LifeCycle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class RESTCatalogServer implements LifeCycle.Listener {
+  private static final Logger LOG = LoggerFactory.getLogger(RESTCatalogServer.class);
+
+  static final String REST_PORT = "rest.port";
+  static final int REST_PORT_DEFAULT = 8181;
+
+  private Server httpServer;
+
+  RESTCatalogServer() {}
+
+  static class CatalogContext {
+    private final Catalog catalog;
+    private final Map<String, String> configuration;
+
+    CatalogContext(Catalog catalog, Map<String, String> configuration) {
+      this.catalog = catalog;
+      this.configuration = configuration;
+    }
+
+    public Catalog getCatalog() {
+      return catalog;
+    }
+
+    public Map<String, String> getConfiguration() {
+      return configuration;
+    }
+  }
+
+  private CatalogContext initializeBackendCatalog() throws IOException {
+    // Translate environment variables to catalog properties
+    Map<String, String> catalogProperties = RCKUtils.environmentCatalogConfig();
+
+    // Fallback to a JDBCCatalog impl if one is not set
+    catalogProperties.putIfAbsent(CatalogProperties.CATALOG_IMPL, JdbcCatalog.class.getName());
+    catalogProperties.putIfAbsent(CatalogProperties.URI, "jdbc:sqlite::memory:");
+    catalogProperties.putIfAbsent("jdbc.schema-version", "V1");
+
+    // Configure a default location if one is not specified
+    String warehouseLocation = catalogProperties.get(CatalogProperties.WAREHOUSE_LOCATION);
+
+    if (warehouseLocation == null) {
+      File tmp = java.nio.file.Files.createTempDirectory("iceberg_warehouse").toFile();
+      tmp.deleteOnExit();
+      warehouseLocation = tmp.toPath().resolve("iceberg_data").toFile().getAbsolutePath();
+      catalogProperties.put(CatalogProperties.WAREHOUSE_LOCATION, warehouseLocation);
+
+      LOG.info("No warehouse location set.  Defaulting to temp location: {}", warehouseLocation);
+    }
+
+    LOG.info("Creating catalog with properties: {}", catalogProperties);
+    return new CatalogContext(
+        CatalogUtil.buildIcebergCatalog("rest_backend", catalogProperties, new Configuration()),
+        catalogProperties);
+  }
+
+  public void start(boolean join) throws Exception {
+    CatalogContext catalogContext = initializeBackendCatalog();
+
+    RESTCatalogAdapter adapter = new RESTServerCatalogAdapter(catalogContext);
+    RESTCatalogServlet servlet = new RESTCatalogServlet(adapter);
+
+    ServletContextHandler context = new ServletContextHandler(ServletContextHandler.NO_SESSIONS);
+    context.setContextPath("/");
+    ServletHolder servletHolder = new ServletHolder(servlet);
+    servletHolder.setInitParameter("javax.ws.rs.Application", "ServiceListPublic");
+    context.addServlet(servletHolder, "/*");
+    context.setVirtualHosts(null);
+    context.insertHandler(new GzipHandler());
```

Comment:
Setting up the Server/Servlet was introduced by #5781 and then copy/pasted by me to the view tests. @amogh-jahagirdar you're right with your theory. I slimmed down the configuration to 
```
ServletContextHandler context = new ServletContextHandler(ServletContextHandler.NO_SESSIONS);
ServletHolder servletHolder = new ServletHolder(new RESTCatalogServlet(new RESTServerCatalogAdapter(catalogContext)));
context.addServlet(servletHolder, "/*");
```
and tests still pass. I guess it would be nice to at least run with GZIP though. I'll probably open a separate PR to slim down the configuration for `TestRESTCatalog` / `TestRESTViewCatalog`.

Reply:
opened https://github.com/apache/iceberg/pull/10945

Reply:
Yeah, it would be great to slim down, but I do think we need to support gzip since this is used as the test backend server and we should handle compressed client request/responses.

Reply:
I've removed some of the configs that don't appear to be necessary (it's been a long time since it was created, so I don't recall the reasoning behind them).  I'm not collapsing some of the nested constructors because it just creates wrapping issues without really helping with readability.